src/node_sqlite.cc

```diff
@@ -356,51 +413,18 @@ void UserDefinedFunction::xFunc(sqlite3_context* ctx,
 
   for (int i = 0; i < argc; ++i) {
     sqlite3_value* value = argv[i];
-    MaybeLocal<Value> js_val;
-
-    switch (sqlite3_value_type(value)) {
-      case SQLITE_INTEGER: {
-        sqlite3_int64 val = sqlite3_value_int64(value);
-        if (self->use_bigint_args_) {
-          js_val = BigInt::New(isolate, val);
-        } else if (std::abs(val) <= kMaxSafeJsInteger) {
-          js_val = Number::New(isolate, val);
-        } else {
-          // Ignore the SQLite error because a JavaScript exception is being
-          // thrown.
-          self->db_->SetIgnoreNextSQLiteError(true);
-          sqlite3_result_error(ctx, "", 0);
-          THROW_ERR_OUT_OF_RANGE(isolate,
-                                 "Value is too large to be represented as a "
-                                 "JavaScript number: %" PRId64,
-                                 val);
-          return;
-        }
-        break;
-      }
-      case SQLITE_FLOAT:
-        js_val = Number::New(isolate, sqlite3_value_double(value));
-        break;
-      case SQLITE_TEXT: {
-        const char* v =
-            reinterpret_cast<const char*>(sqlite3_value_text(value));
-        js_val = String::NewFromUtf8(isolate, v).As<Value>();
-        break;
-      }
-      case SQLITE_NULL:
-        js_val = Null(isolate);
-        break;
-      case SQLITE_BLOB: {
-        size_t size = static_cast<size_t>(sqlite3_value_bytes(value));
-        auto data = reinterpret_cast<const uint8_t*>(sqlite3_value_blob(value));
-        auto store = ArrayBuffer::NewBackingStore(isolate, size);
-        memcpy(store->Data(), data, size);
-        auto ab = ArrayBuffer::New(isolate, std::move(store));
-        js_val = Uint8Array::New(ab, 0, size);
-        break;
-      }
-      default:
-        UNREACHABLE("Bad SQLite value");
+    MaybeLocal<Value> js_val =
+        SQLiteValueToJS(isolate, self->use_bigint_args_, value);
+
+    if (js_val.IsEmpty()) {
+      // Ignore the SQLite error because a JavaScript exception is pending.
+      self->db_->SetIgnoreNextSQLiteError(true);
+      sqlite3_result_error(ctx, "", 0);
+      THROW_ERR_OUT_OF_RANGE(isolate,
```

Comment:
I think this should be thrown from inside of the macro. The `from` parameter of the macro can be used to call the proper sqlite function.

Reply:
I think it would be good to have it inside the macro. The thing is, the message is different in each place.

We would end up in the same case as https://github.com/nodejs/node/pull/57571/files#r2008833633

Reply:
I would personally be OK with reusing the same error message.

Reply:
Done